design/specs/visual-query-builder.md

```diff
@@ -0,0 +1,192 @@
+
+#### From a Table (Sidebar Navigation or Open Table)
+
+- From an open table, a user clicks the `Create View from [Table Name]` button located in the toolbar. This action will open `Data Explorer` and automatically set the table as `Base Table` and select all direct columns.
+- In the `Schema Browser` view, a user opens the contextual menu for a table navigation item and selects `Create view from Table`. This action launches `Data Explorer` and automatically sets the selected table as `Base Table` and adds its direct columns to a `Select Column` step.
+- The system will also add link columns if present in the `Base Table`. However, it won't include link columns referencing the `Base Table` from other tables. The system will set the values for link columns to the referenced table's primary key by default.
+- The user can then choose to remove or add additional columns.
+
+### User opens the Data Explorer to inspect an existing view
+
+#### From an open view
+
+- From an active view tab, a user can inspect a view by clicking on the `Open in Data Explorer` button in the toolbar area.
+- Views need to use supported query commands and functions to be inspected in `Data Explorer` with full functionality. When queries are not supported, an option to see the underlying query will be available.
+
+#### From a view navigation item
+
+- In the `Schema Browser` view, a user opens the contextual menu for a view navigation item and selects the option `Open in Data Explorer`. This action will open `Data Explorer` and display the properties of the selected view.
+
+## Base Table
+
+Views require a base table to be set so that the available columns and links can be listed for selection.
+
+### User sets the base table for a new view
+
+- For users starting a new view from scratch, `Base Table` selection will be the first required step.
+- Once `Data Explorer` is launched the user will be prompted to make a selection. The system will list all tables from the active schema. See *Wireframe 1* for an example of the base table selection prompt.
+- Once the `Base Table` is selected, a second step for `Select Columns` is automatically added and the user is prompted to continue with the view creation process.
+- At this point, the result table is empty.
+
+[Wireframe 1](https://share.balsamiq.com/c/jp7WFu663j6sbfKNknuq5Z.png)
+
+### User inspects the base table for an existing view in Data Explorer
+
+- When inspecting an existing view, a user can find the `Base Table` as the first step of the `Workflow Sidebar`.
+- The `Base Table` step cannot be removed, only edited, causing all subsequent steps to be cleared.
+
+[Wireframe 2](https://share.balsamiq.com/c/wTEzQcBamoBBWBSSF5Cmuy.png)
+
+### User opens a view where the base table is missing
+
+- If a user deletes the base table for a view, they should be able to open the view in `Data Explorer` and either resolve the error by replacing the base table with an identical one or clear the workflow and start over.
```

Comment:
This scenario will never happen. If the user tries to delete the base table, depending on the options, it will either:
- refuse to delete the base table because it has dependent objects (i.e. the view)
- delete the view along with the table.

Reply:
@kgodey what if the base table was deleted outside of Mathesar?